modules/react-components/src/components/renderer/markdown/components/blockquote.tsx

```diff
@@ -0,0 +1,120 @@
+import { MarkdownCustomComponentPropsInterface } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement } from "react";
+import { childRenderer } from "./utils";
+
+/**
+ * Custom blockquote component types.
+ */
+enum CustomBlockquoteTypes {
+    WRAPPER = "wrapper"
+}
+
+/**
+ * Props interface for the `Blockquote` component.
+ */
+interface BlockquoteProps extends MarkdownCustomComponentPropsInterface<"blockquote"> {
+    /**
+     * Custom attributes supplied by the 'rehype-attr' plugin.
+     */
+    "data-config"?: {
+        /**
+         * Type of the alert message.
+         */
+        type?: AlertProps["severity"] | CustomBlockquoteTypes;
+        /**
+         * Title of the alert message.
+         */
+        title?: string;
+        /**
+         * Variant of the alert box.
+         * filled or outlined
+         */
+        variant?: AlertProps["variant"];
+        /**
+         * Boolean flag to determine if the icon should be displayed.
+         */
+        icon?: boolean;
+        /**
+         * If the component type is wrapper how much indentation is needed.
+         * 1 Indent = 5px
+         */
+        indent?: number;
+    };
+}
+
+/**
+ * Markdown custom component for the blockquote element.
+ *
+ * @param Props - Props to be injected into the component.
+ */
+const Blockquote: FunctionComponent<BlockquoteProps> = (props: BlockquoteProps): ReactElement => {
+    const {
+        children,
+        "data-config": dataConfig,
+        "data-componentid": componentId
+    } = props;
+
+    if (!children || !Array.isArray(children)) {
+        return null;
+    }
+
+    return (
+        dataConfig?.type === CustomBlockquoteTypes.WRAPPER || !dataConfig?.type
+            ? (
+                <div
+                    style={ {
+                        marginLeft: `${ 5 * (dataConfig?.indent === undefined ? 9 : dataConfig?.indent) }px`
+                    } }
+                >
+                    { childRenderer(props) }
+                </div>
+            )
+            : (
+                <Alert
+                    severity={ dataConfig?.type || "info" }
+                    icon={ dataConfig?.icon === false ? false : undefined }
+                    variant={ dataConfig?.variant || "standard" }
+                    data-componentid={ componentId }
+                    style={ {
+                        marginLeft: `${ 5 * (dataConfig?.indent === undefined ? 0 : dataConfig?.indent) }px`
+                    } }
+                >
+                    {
+                        dataConfig?.title ? (
+                            <AlertTitle>{ dataConfig?.title }</AlertTitle>
+                        ) : null
+                    }
+                    { childRenderer(props) }
+                </Alert>
+            )
+    );
+};
+
+/**
+ * Default props for the `Blockquote` component.
+ */
+Blockquote.defaultProps = {
+    "data-componentid": "custom-markdown-blockquote"
```

Comment:
Let's refrain from specifying defaultProps for new components, as it's going to be deprecated. Please check and update the other places as well.

https://github.com/facebook/react/pull/16210

Reply:
https://github.com/wso2/identity-apps/pull/5896/commits/2253e2e1733aa73e5e987bb67f7148c8a31c58a9